lib/txmgr/txmgr.go

```diff
@@ -495,16 +484,25 @@ func (m *SimpleTxManager) queryReceipt(ctx context.Context, txHash common.Hash,
 	sendState *SendState) (*types.Receipt, bool, error) {
 	ctx, cancel := context.WithTimeout(ctx, m.cfg.NetworkTimeout)
 	defer cancel()
+
+	// there are no receipts for pending transactions, therefore check if the
+	// transaction is pending first and only then proceed
+	_, pending, err := m.backend.TransactionByHash(ctx, txHash)
```

Comment:
See [geth code](https://github.com/ethereum/go-ethereum/blob/fe91d476ba3e29316b6dc99b6efd4a571481d888/core/blockchain_reader.go#L282), this call can also return "transaction indexing still in progress".